cmd/crank/build.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/spf13/afero"
+
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+
+	"github.com/crossplane/crossplane/pkg/xpkg"
+)
+
+// buildCmd builds a package.
+type buildCmd struct {
+	Configuration buildConfigCmd   `cmd:"" help:"Build a Configuration package."`
+	Provider      buildProviderCmd `cmd:"" help:"Build a Provider package."`
+
+	PackageRoot string   `short:"f" help:"Path to crossplane.yaml" default:"."`
+	Ignore      []string `name:"ignore" help:"Paths, specified relative to --package-root, to exclude from the package."`
+}
+
+// Run runs the build cmd.
+func (c *buildCmd) Run(child *childArg) error {
```

Comment:
Note: this does not currently lint the package while building, I would prefer to introduce that and the stand-alone `lint` command as well in a separate PR that includes switching to https://github.com/crossplane/crossplane-runtime/pull/211 since this PR is already so big.